dataproc/dataproc.go

```diff
@@ -0,0 +1,313 @@
+	"golang.org/x/oauth2/google"
+
+	dataproc "google.golang.org/api/dataproc/v1"
+	storage "cloud.google.com/go/storage"
+)
+
+const (
+	// Filename of example job (Python) file uploaded to Cloud Storage
+	exampleJobName = "test-file"
+
+	// Allows for full access to Google Cloud Platform products
+	scope = "https://www.googleapis.com/auth/cloud-platform"
+
+	// Compute URI base
+	computeUriBase = "https://www.googleapis.com/compute/v1/projects/%s/zones/%s"
+)
+
+var (
+	bucketName  = flag.String("bucket", "", "GCS Bucket for storage of job file")
+	clusterName = flag.String("cluster-name", "", "Name of cluster")
+	projectID   = flag.String("project", "", "Your cloud project ID.")
+	pysparkFile = flag.String("pyspark-file", "", "PySpark Job file")
+	region      = flag.String("region", "global", "Your cloud project region.")
+	zoneID      = flag.String("zone", "", "Cloud Platform zone")
+)
+
+func main() {
+	// Parse command line arguments
+	flag.Parse()
+
+	// Check for required flags (golang flags does not support?)
+	if *bucketName == "" || *clusterName == "" || *projectID == "" || *pysparkFile == "" || *region == "" || *zoneID == "" {
+		log.Fatal("Incorrect arguments specified see 'go-dataproc -help' for help")
+	}
+
+	// Create a new Dataproc service
+	service, err := getDataprocService()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new storage service
+	storageService, err := getStorageService()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a cluster
+	_, err = createCluster(service, *clusterName, *region, *projectID, *zoneID)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Cluster created")
+
+	// Wait on the cluster to become active
+	fmt.Println("Waiting for cluster to be ready")
+	_, err = waitForCluster(service, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Submit a job to the cluster
+	fmt.Println("Submitting job")
+	jobId, err := submitJob(service, storageService, *pysparkFile, *projectID, *bucketName, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Wait for the job to complete
+	fmt.Println("Waiting for job to complete")
+	_, err = waitForJob(service, jobId, *projectID, *region)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Job is finished")
+
+	// Get the job output
+	clusterData, err := getClusterDataByName(service, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	output, err := getJobOutput(storageService, *projectID, clusterData[1], clusterData[3], jobId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Job output:\n%s\n", string(output))
+
+	// Delete the cluster
+	_, err = deleteCluster(service, *projectID, *region, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Cluster deleted")
+	fmt.Printf("Done...")
+}
+
+// getDataprocService creates and returns a Cloud Dataproc service object.
+func getDataprocService() (service *dataproc.Service, err error) {
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, scope)
```

Comment:
This error is ignored. I suggest inlining this (move all of this up to `main()`) too.